concrete/src/Page/Collection/Version/GlobalVersionList.php

```diff
@@ -0,0 +1,84 @@
+<?php
+namespace Concrete\Core\Page\Collection\Version;
+
+use Concrete\Core\Search\ItemList\Database\ItemList as DatabaseItemList;
+use Concrete\Core\Search\StickyRequest;
+
+/**
+ * An object that holds a list of collection versions.
+ *
+ * If you need a list of versions of multiple collections / pages,
+ * use this class. Otherwise use the VersionList class.
+ */
+class GlobalVersionList extends DatabaseItemList
+{
+    /**
+     * @param \Concrete\Core\Search\StickyRequest|null $req
+     */
+    public function __construct(StickyRequest $req = null)
+    {
+        parent::__construct($req);
+    }
+
+    /**
+     * @param array $queryRow
+     *
+     * @return \Concrete\Core\Page\Collection\Version\Version
+     */
+    public function getResult($queryRow)
+    {
+        $version = new Version();
+        $version->setPropertiesFromArray($queryRow);
+
+        return $version;
+    }
+
+    public function createQuery()
+    {
+        $this->query->select('cv.*')
+            ->from('CollectionVersions', 'cv');
+    }
+
+    /**
+     * @return int
+     */
+    public function getTotalResults()
+    {
+        return (int) $this->query
```

Comment:
Shouldn't we clone the class `query` property instead of changing it?

Reply:
Yes, we should. Fixed in 15582516a3c23b90fd3733ca4eeea97e33d70a93